client/js/requests.js

```diff
@@ -0,0 +1,43 @@
+async function getAll(){
+    try {
+        const response = await fetch(`http://localhost:3000/posts/`);
+        const data = await response.json()
+        return data;
+    } catch (err) {
+        console.warn(err);
+    }
+}
+
+async function getItem(id) {
+    try {
+        const response = await fetch(`http://localhost:3000/posts/${id}`);
+        const data = await response.json();
+        return data;
+    } catch (err) {
+        console.warn(err);
+    }
+}
+
+async function makePost(e){
+    e.preventDefault();
+    try {
+        const options = {
+            method: 'POST',
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify(Object.fromEntries(new FormData(e.target)))
+        }
+        
+        const response = await fetch('http://localhost:3000/posts/', options); 
+        const { id, err } = await response.json();
+        if(err) { 
+            throw Error(err) 
+        } else {
+            window.location.hash = `#posts/${id}`
```

Comment:
Im assuming you used the client from assignment and the completed database exercise, the javascript files look a bit similar to them but honestly, I don't know what Beth has done so we didnt use those, some sections I understand whats going on and some I don't so I didnt want to comment too much on the functionality of the code.